gwpy/timeseries/timeseries.py

```diff
@@ -1290,6 +1290,63 @@ def rms(self, stride=1):
         return self.__class__(data, channel=self.channel, t0=self.t0,
                               name=name, sample_rate=(1/float(stride)))
 
+    def demodulate(self, f, stride=1, exp=False, deg=True):
+        """Compute the average magnitude and phase of this `TimeSeries`
+           once per stride at a given frequency.
+
+        Parameters
+        ----------
+        f : `float`
+            frequency (Hz) at which to demodulate the signal
+
+        stride : `float`, optional
+            stride (seconds) between calculations, defaults to 1 second
+
+        exp : `bool`, optional
+            return the demodulated magnitude and phase trends as one
+            `TimeSeries` object representing a complex exponential
+
+        deg : `bool`, optional
+            if `exp=False`, calculates the phase in degrees
+
+        Returns
+        -------
+        mag, phase : `TimeSeries`
+            if `exp=False`, returns a pair of `TimeSeries` objects representing
+            magnitude and phase trends with `dt=stride`
+
+        out : `TimeSeries`
+            if `exp=True`, returns a single `TimeSeries` with magnitude and
+            phase trends represented as `mag * exp(1j*phase)` with `dt=stride`
+        """
+        stridesamp = int(stride * self.sample_rate.value)
+        nsteps = int(self.size // stridesamp)
+        # mix with a complex oscillator and stride through the TimeSeries,
+        # taking the average over each stride
+        out = numpy.zeros(nsteps, dtype=complex).view(type(self))
+        out.__metadata_finalize__(self)
+        out.sample_rate = 1/float(stride)
+        out._unit = self.unit
+        mixed = 2 * numpy.exp(-2*numpy.pi*1j*f*self.times.value) * self.value
+        for step in range(nsteps):
+            # find step TimeSeries
+            idx = int(stridesamp * step)
+            idx_end = idx + stridesamp
+            stepseries = mixed[idx:idx_end]
+            demod_ = numpy.average(stepseries)
+            out.value[step] = demod_
+        if exp:
+            return out
+        else:
```

Comment:
@alurban, since the `if` statement includes `return`, you don't need the `else` at all. Also, I think I prefer using `override_unit()` compared to `_unit` (using a ternary instead of `if/else`).

If you agree, can you please modify this block to something like:

```python
        if exp:
            return out
        mag = numpy.abs(out)
        phase = numpy.angle(out, deg=deg).view(type(self))
        phase.__metadata_finalize__(out)
        phase.override_unit('deg' if deg else 'rad')
        return mag, phase